### DISCUSSION TITLE
How to use summaries and datetimes.

### DISCUSSION THREAD

**Discussion by billgeo** [radiantearth/stac-spec#1156] [General STAC Q&A] [answered]
Hi, we are looking at adding some timestamps to our collection.json that indicate the dates that the collection was first created and the date that the collection was last updated. We're interested in timestamps for the asset files and also the metadata files.

We looked at the timestamps extension which looks like what we want but the examples are only for items. Can someone help with how we should implement this at the collection level? It sounds like we should summarise (min and max) the item values. But not sure how to do that and differentiate between the different types of timestamps.

Here's the timestamps we're interested from an item.json example.
Created and updated times for the metadata itself: https://github.com/stac-extensions/timestamps/blob/1b43bcc1d823b4bdd69ff962cac412ec6ef07829/examples/item.json#L43-L44
Created and updated times for the assets: https://github.com/stac-extensions/timestamps/blob/1b43bcc1d823b4bdd69ff962cac412ec6ef07829/examples/item.json#L83-L84

**Comment by m-mohr** [comment, marked as answer]
Hi @billgeo,

summarizing Item Properties works as you have described it. You could add something like the following to the Collection:
```
"summaries": {
  "created": {
    "minimum": "2018-10-02T00:00:03Z",
    "maximum": "2019-10-02T00:00:03Z"
  },
  "updated": {
    "minimum": "2020-05-20T12:13:02Z",
    "maximum": "2021-05-20T12:13:02Z"
  }
}
```
Note: this describes/summarizes the Item creation/update timestamps. To add the creation/update time of the collection metadata, just add the `updated`/`created` fields to the Collection (top-level).

To give specific values for Item Assets there's the Item Asset Definition extension:
https://github.com/stac-extensions/item-assets
Unfortunately, it doesn't provide a way to summarize those values.
There is a related open issue here:
https://github.com/radiantearth/stac-spec/issues/1153

HTH!

**Reply by billgeo** [reply]
Thank @m-mohr . That's great to know. We'll use it that way. And watch the open issue.
